src/openxr/OpenXRApi.cpp

```diff
@@ -631,34 +672,44 @@ bool OpenXRApi::initialiseInstance() {
 	// Damn you microsoft for not supporting this!!
 	// XrExtensionProperties extensionProperties[extensionCount];
 	XrExtensionProperties *extensionProperties = (XrExtensionProperties *)malloc(sizeof(XrExtensionProperties) * extensionCount);
-	if (extensionProperties == NULL) {
+	if (extensionProperties == nullptr) {
 		Godot::print_error("OpenXR Couldn't allocate memory for extension properties", __FUNCTION__, __FILE__, __LINE__);
 		return false;
 	}
-	for (uint16_t i = 0; i < extensionCount; i++) {
+	for (uint32_t i = 0; i < extensionCount; i++) {
 		extensionProperties[i].type = XR_TYPE_EXTENSION_PROPERTIES;
-		extensionProperties[i].next = NULL;
+		extensionProperties[i].next = nullptr;
 	}
 
-	result = xrEnumerateInstanceExtensionProperties(NULL, extensionCount, &extensionCount, extensionProperties);
+	result = xrEnumerateInstanceExtensionProperties(nullptr, extensionCount, &extensionCount, extensionProperties);
 	if (!xr_result(result, "Failed to enumerate extension properties")) {
 		free(extensionProperties);
 		return false;
 	}
 
 #ifdef ANDROID
-	if (!isExtensionSupported(XR_KHR_OPENGL_ES_ENABLE_EXTENSION_NAME, extensionProperties, extensionCount)) {
-		Godot::print_error("OpenXR Runtime does not support OpenGLES extension!", __FUNCTION__, __FILE__, __LINE__);
-		free(extensionProperties);
-		return false;
-	}
-
-	if (!isExtensionSupported(XR_KHR_ANDROID_CREATE_INSTANCE_EXTENSION_NAME, extensionProperties, extensionCount)) {
-		Godot::print_error("OpenXR Runtime does not support android instance extension!", __FUNCTION__, __FILE__, __LINE__);
-		free(extensionProperties);
-		return false;
+	// Check that the extensions required are present.
```

Comment:
Hmm, this leaves me wondering if we shouldn't build a vector with required extensions, then populate them according to platform or other switches, and have the rest of the logic the same for all platforms.

Reply:
On that subject, some of these should be checked regardless and we simply check whether the extension is available or not. If it is available we make use of it, if not we should act accordingly. For instance `XR_FB_DISPLAY_REFRESH_RATE_EXTENSION_NAME` not being available isn't severe enough to fail initialising the plugin, it just means we can't query nor set the refresh rate. But as more vendors implement this extension it becomes usable in those runtimes that make it available.

Reply:
Let me address this in another PR. I want to see a mechanism for required extensions, and another for optional ones which would trigger a flag based on availability.
I'll add a `TODO` for the time being.